tests/commonTemplates_test.go

```diff
@@ -208,6 +211,42 @@ var _ = Describe("Common templates", func() {
 	})
 
 	Context("resource change", func() {
+		Context("namespace change", func() {
+			BeforeEach(func() {
+				Eventually(func() error {
+					namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: strategy.GetNewTemplatesNamespace()}}
+					return apiClient.Create(ctx, namespaceObj)
```

Comment:
Instead of  setting `Name: strategy.GetNewTemplateNamespace()`, can you set `GenerateName` to some constant? It will create a namespaces with a random suffix.

Then you can remove `GetNewTemplateNamespace()` from `strategy`. The methods in `strategy` are used to get values that are configurable from outside of the test script, using environment variables, and the new template namespace is not configurable.

And if you want, you can remove the `Eventually` block and create the namespace directly.